chain/beacon/sync_manager.go

```diff
@@ -153,11 +167,12 @@ func (s *SyncManager) Run() {
 			// must have gotten some data.
 			upperBound := lastRoundTime + int(s.period.Seconds())*s.factor
 			if upperBound < int(s.clock.Now().Unix()) {
+				span.End()
 				// we haven't received a new block in a while
 				// -> time to start a new sync
 				cancel()
-				ctx, cancel = context.WithCancel(context.Background())
-				//nolint
+				ctx, cancel = context.WithCancel(s.ctx)
+				//nolint:errcheck // TODO: Handle this
```

Comment:
don't forget to handle this!

Reply:
We need to discuss about this. The reason for this is, the call is async. We should rewrite most of the logic there to:
- first wait for the existing `s.Sync()` call to finish after a cancel is issued
- then, start a new `s.Sync()`.
- finally, if an error happens, we need to determine how to handle the error: just log it, abort the SyncManager, or something else?

Reply:
That's a good point, we should wait for the previous one to finish. But that should be fairly quick I guess, since we can close the pipes from our side directly.

Errors in the Sync manager should only be logged but not abort the whole thing. We could have an exponential cool-off period to avoid DOSing our node while trying to query the other ones in case of an issue with all other nodes I guess.